fastcrypto-zkp/src/unit_tests/api_tests.rs

```diff
@@ -1,171 +1,113 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
+
 use crate::api::{prepare_pvk_bytes, verify_groth16_in_bytes};
+use crate::dummy_circuits::DummyCircuit;
+use crate::verifier::process_vk_special;
+use ark_bls12_381::{Bls12_381, Fr};
+use ark_crypto_primitives::snark::SNARK;
+use ark_groth16::Groth16;
+use ark_serialize::CanonicalSerialize;
+use ark_std::rand::thread_rng;
+use ark_std::UniformRand;
+use std::ops::Mul;
 
 #[test]
 fn test_verify_groth16_in_bytes_api() {
+    const PUBLIC_SIZE: usize = 128;
+    let rng = &mut thread_rng();
+    let c = DummyCircuit::<Fr> {
+        a: Some(<Fr>::rand(rng)),
+        b: Some(<Fr>::rand(rng)),
+        num_variables: PUBLIC_SIZE,
+        num_constraints: 10,
+    };
+
+    let (pk, vk) = Groth16::<Bls12_381>::circuit_specific_setup(c, rng).unwrap();
+    let proof = Groth16::<Bls12_381>::prove(&pk, c, rng).unwrap();
+    let v = c.a.unwrap().mul(c.b.unwrap());
+    let blst_pvk = process_vk_special(&vk);
+
+    let bytes = blst_pvk.as_serialized().unwrap();
+
+    let mut proof_inputs_bytes = vec![];
+    v.serialize_compressed(&mut proof_inputs_bytes).unwrap();
+
+    let mut proof_points_bytes = vec![];
+    proof.serialize_compressed(&mut proof_points_bytes).unwrap();
+
     // Success case.
     assert!(verify_groth16_in_bytes(
-        &VK_GAMMA_ABC_G1_VECTOR,
-        &ALPHA_G1_BETA_G2_BYTES,
-        &GAMMA_G2_NEG_PC_BYTES,
-        &DELTA_G2_NEG_PC,
-        &PROOF_INPUTS_BYTES,
-        &PROOF_POINTS_BYTES
+        &bytes[0],
```

Comment:
why did we change the name of the variables here from descriptive to bytes[]?

Reply:
That was just a convenience, but I see why it is a bit unclear. I've added a pr to fix this.